hikari/channels.py

```diff
@@ -812,3 +813,37 @@ class GuildVoiceChannel(GuildChannel):
 
     If this is `0`, then assume no limit.
     """
+
+
+@attr.s(eq=True, hash=False, init=False, slots=False)
+class UnrecognisedChannel(PartialChannel):
+    """Channels that haven't been implemented in the library.
+
+    The attributes on this object are undocumented and dynamic.
+
+    Example
+    -------
+        >>> entry_info.foobar.baz
+
+    !!! note
+        This model has no slots and will have arbitrary undocumented attributes
+        (in it's `__dict__` based on the received payload).
+        The only attributes that are always garantied to be there are `app`,
+        `id`, `name` and `type`.
```

Comment:
What about a nullable guild id attribute

Reply:
I rather keep it with only the same attributes as Partial Channel